neqo-udp/src/lib.rs

```diff
@@ -28,184 +28,123 @@ std::thread_local! {
     static RECV_BUF: RefCell<Vec<u8>> = RefCell::new(vec![0; RECV_BUF_SIZE]);
 }
 
-/// A UDP socket.
-pub struct Socket<S> {
-    state: UdpSocketState,
-    inner: S,
+pub fn send_inner(
+    state: &UdpSocketState,
+    socket: quinn_udp::UdpSockRef<'_>,
+    d: &Datagram,
+) -> io::Result<()> {
+    let transmit = Transmit {
+        destination: d.destination(),
+        ecn: EcnCodepoint::from_bits(Into::<u8>::into(d.tos())),
+        contents: d,
+        segment_size: None,
+        src_ip: None,
+    };
+
+    state.send(socket, &transmit)?;
+
+    qtrace!(
+        "sent {} bytes from {} to {}",
+        d.len(),
+        d.source(),
+        d.destination()
+    );
+
+    Ok(())
 }
 
-impl<S> Socket<S> {
-    fn send_inner(
```

Comment:
Not removed. Just moved from a method on `Socket` to a free function, i.e. reduced the indentation by 1. Allows `neqo_bin::udp` to call them.